debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/infinispan/InfinispanTransactionCache.java

```diff
@@ -11,34 +11,33 @@
 import org.infinispan.commons.util.CloseableIterator;
 
 import io.debezium.connector.oracle.Scn;
-import io.debezium.connector.oracle.logminer.events.Transaction;
 import io.debezium.connector.oracle.logminer.processor.TransactionCache;
 
 /**
  * A {@link TransactionCache} implementation for use with embedded Infinispan.
  *
  * @author Chris Cranford
  */
-public class InfinispanTransactionCache implements TransactionCache<Cache.Entry<String, Transaction>> {
+public class InfinispanTransactionCache implements TransactionCache<InfinispanTransaction, Cache.Entry<String, InfinispanTransaction>> {
 
-    private final Cache<String, Transaction> cache;
+    private final Cache<String, InfinispanTransaction> cache;
```

Comment:
So aren't we still storing all the events from one TX within a single value of the cache? I'm confused.

Reply:
No, only the memory implementation does this.

For Infinispan, the processor has a new event cache where we store the events separately from the TX.  The `InfinispanTransaction` now only captures transaction metadata such as the ID, its starting SCN, the time when the transaction started, and the high watermark number of events that participate in that TX.

Reply:
https://github.com/debezium/debezium/blob/7799aba3f2ee1e1f316646b4d804d2f3d8e82cfd/debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/infinispan/InfinispanLogMinerEventProcessor.java#L68-L73